src/core/SIP/SIPAuthorisationDigest.cs

```diff
@@ -85,69 +85,103 @@ public static SIPAuthorisationDigest ParseAuthorisationDigest(SIPAuthorisationHe
-                        {
-                            authRequest.Realm = headerValue;
-                        }
-                        else if (Regex.Match(headerName, "^" + AuthHeaders.AUTH_NONCE_KEY + "$", RegexOptions.IgnoreCase).Success)
-                        {
-                            authRequest.Nonce = headerValue;
-                        }
-                        else if (Regex.Match(headerName, "^" + AuthHeaders.AUTH_USERNAME_KEY + "$", RegexOptions.IgnoreCase).Success)
-                        {
-                            authRequest.Username = headerValue;
-                        }
-                        else if (Regex.Match(headerName, "^" + AuthHeaders.AUTH_RESPONSE_KEY + "$", RegexOptions.IgnoreCase).Success)
-                        {
-                            authRequest.Response = headerValue;
-                        }
-                        else if (Regex.Match(headerName, "^" + AuthHeaders.AUTH_URI_KEY + "$", RegexOptions.IgnoreCase).Success)
+                        static string BuildQop(ReadOnlySpan<char> headerValue)
                         {
-                            authRequest.URI = headerValue;
+                            Span<char> chars = stackalloc char[headerValue.Length];
+                            _ = headerValue.ToLowerInvariant(chars);
+                            return new string((ReadOnlySpan<char>)chars);
                         }
-                        else if (Regex.Match(headerName, "^" + AuthHeaders.AUTH_CNONCE_KEY + "$", RegexOptions.IgnoreCase).Success)
-                        {
-                            authRequest.Cnonce = headerValue;
-                        }
-                        else if (Regex.Match(headerName, "^" + AuthHeaders.AUTH_NONCECOUNT_KEY + "$", RegexOptions.IgnoreCase).Success)
-                        {
-                            Int32.TryParse(headerValue, out authRequest.NonceCount);
-                        }
-                        else if (Regex.Match(headerName, "^" + AuthHeaders.AUTH_QOP_KEY + "$", RegexOptions.IgnoreCase).Success)
-                        {
-                            authRequest.Qop = headerValue.ToLower();
-                        }
-                        else if (Regex.Match(headerName, "^" + AuthHeaders.AUTH_OPAQUE_KEY + "$", RegexOptions.IgnoreCase).Success)
+#else
+                        authRequest.Qop = headerValue.ToString().ToLowerInvariant();
+#endif
+                    }
+                    else if (headerName.Equals(AuthHeaders.AUTH_OPAQUE_KEY.AsSpan(), StringComparison.OrdinalIgnoreCase))
+                    {
+                        authRequest.Opaque = headerValue.ToString();
+                    }
+                    else if (headerName.Equals(AuthHeaders.AUTH_ALGORITHM_KEY.AsSpan(), StringComparison.OrdinalIgnoreCase))
+                    {
+                        //authRequest.Algorithm = headerValue;
+
+#if NETCOREAPP6_0_OR_GREATER

```

Comment:
Is this preprocessor define correct? It's not lsied on https://learn.microsoft.com/en-us/dotnet/standard/frameworks#preprocessor-symbols.

Reply:
The '-' are only being replaced because they can't be used in C# enums. It's safe to do so for the AUTH_ALGORITHM_KEY values. If the algorithm is not in the enum it means it's not recognised and the digest won't be able to be verified/generated any way,

Reply:
Other than `MD5` and `SHA256` what other values are to be expected? `SHA-256`?

I wouldn't expect `-M-D-5-` to be valid. Is it?

Reply:
MD5
SHA-256
SHA-512 (not wired up, although would be trivial)

Reply:
Is `SHA256` a valid value? Is it case insensitive?